### DISCUSSION TITLE
How can I make packages I publish visible to other members of the organization by default?

### DISCUSSION THREAD

**Discussion by wangyizhi1** [community/community#66098] [Packages] [answered]
### Select Topic Area

Question

### Body

Hello,

I have a question regarding Docker image publishing on GitHub Packages. When I push an image, the package is set to private by default, making it invisible to other members of the organization. This forces me to manually change the package's visibility on the web page each time. Is there a straightforward method to make my packages automatically public every time I push an image? Alternatively, is there a way to grant visibility of private packages to other organization members without manual intervention?

I appreciate any guidance and assistance with this matter. Thank you in advance!

Best regards!

**Comment by arafajs** [comment, marked as answer]
- Set the default visibility for your repository to public. This will make any packages published from that repo public as well. You can do this by going to the repo settings, selecting Options, and changing the repo visibility.
- Use GitHub Actions to automatically publish your packages and set the visibility. In your workflow file, use the `publish` action from `docker/build-push-action` and set `visibility: public`.
- Configure access control for your package separately from the repository. Go to Settings > Packages > Your Package > Manage Access to configure which users/teams can see the package.
- If you want to keep the repo private but allow organization members access, add your organization teams with `read` permissions on the package access control.
- Consider creating a separate public repo just for publishing packages that you want visible. Keep your main development repo private.

**Reply by mkst** [reply]
> Unexpected input(s) 'visibility', valid inputs are ['add-hosts', 'allow', 'attests', 'build-args', 'build-contexts', 'builder', 'cache-from', 'cache-to', 'cgroup-parent', 'context', 'file', 'labels', 'load', 'network', 'no-cache', 'no-cache-filters', 'outputs', 'platforms', 'provenance', 'pull', 'push', 'sbom', 'secrets', 'secret-files', 'shm-size', 'ssh', 'tags', 'target', 'ulimit', 'github-token']

Any ideas?

**Reply by roshkhatri** [reply]
+1 on this

**Reply by douglowe** [reply]
I agree - this answer does not work.
